now/executor/autocomplete/executor.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+
+from better_profanity import profanity
+from docarray import DocumentArray
+from fast_autocomplete import AutoComplete
+
+from now.executor.abstract.auth.auth import NOWAuthExecutor as Executor
+from now.executor.abstract.auth.auth import SecurityLevel, secure_request
+
+
+class NOWAutoCompleteExecutor(Executor):
+    def __init__(self, search_traversal_paths: str = '@r', words=None, *args, **kwargs):
+        self.words = words if words else {}
+        self.search_traversal_paths = search_traversal_paths
+        self.autocomplete = None
+        self.char_threshhold = 100
+
+        super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def contains_profanity(text: str):
+        """Helper function that wraps checking for profanities.
+        :param text: Text to be checked for profanities
+        :returns: Boolean values signifying whether or not profanity
+            was present in the text
+        """
+        return profanity.contains_profanity(text)
+
+    def update_words(self, word):
+        if word in self.words:
+            self.words[word]['count'] += 1
+        else:
+            self.words[word] = {'count': 1}
+
+    @secure_request(on='/search', level=SecurityLevel.USER)
+    def search_update(
+        self, docs: Optional[DocumentArray] = None, parameters: dict = {}, **kwargs
+    ):
+        traversal_paths = parameters.get('traversal_paths', self.search_traversal_paths)
+        flat_docs = docs[traversal_paths]
+
+        for doc in flat_docs:
+            if doc.text and not NOWAutoCompleteExecutor.contains_profanity(doc.text):
```

Comment:
`contains_profinity` returns True if it contains some profanity right? Then shouldn't this check be? :
```suggestion
            if doc.text and NOWAutoCompleteExecutor.contains_profanity(doc.text):
```

Reply:
Ah okay, never mind. I got what this is doing. So you add the searched words if it is not profane one.